### PR TITLE
Update pyproject.toml

### PR DESCRIPTION
**Pull request by Aj-esh**
added dask as known-third-party in pyproject.toml to consider the downloading of dependsncy dask dataframe for needed usecases.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by Innixma** [comment]
@Aj-esh Where is dask used in AutoGluon? What happens if this PR isn't merged and what changes when it is merged?

**Comment by Aj-esh** [comment]
implementations of parallel computations of ensemble  and the implementation is falling back to non-Dask implementation on fitting task.
Add Dask to dependencies list as dask is bring in optimized parallel data processing and it eliminate warnings during model training.
It is enhancing the performance for decently large datasets by good amount

**Comment by Innixma** [comment]
I'll need to see logs so that I better understand what is going on, as to my knowledge we don't use dask in AutoGluon.

**Comment by Aj-esh** [comment]
Verbosity: 2 (Standard Logging)
=================== System Info ===================
AutoGluon Version:  1.2
Python Version:     3.11.11
Operating System:   Linux
Platform Machine:   x86_64
Platform Version:   #1 SMP PREEMPT_DYNAMIC Thu Jun 27 21:05:47 UTC 2024
CPU Count:          2
Memory Avail:       11.48 GB / 12.67 GB (90.6%)
Disk Space Avail:   73.53 GB / 107.72 GB (68.3%)
===================================================
Presets specified: ['high_quality']
Setting dynamic_stacking from 'auto' to True. Reason: Enable dynamic_stacking when use_bag_holdout is disabled. (use_bag_holdout=False)
Stack configuration (auto_stack=True): num_stack_levels=1, num_bag_folds=8, num_bag_sets=1
Note: `save_bag_folds=False`! This will greatly reduce peak disk usage during fit (by ~8x), but runs the risk of an out-of-memory error during model refit if memory is small relative to the data size.
	You can avoid this risk by setting `save_bag_folds=True`.
DyStack is enabled (dynamic_stacking=True). AutoGluon will try to determine whether the input data is affected by stacked overfitting and enable or disable stacking as a consequence.
	This is used to identify the optimal `num_stack_levels` value. Copies of AutoGluon will be fit on subsets of the data. Then holdout validation data is used to detect stacked overfitting.
	Running DyStack for up to 875s of the 3500s of remaining time (25%).
	Running DyStack sub-fit in a ray process to avoid memory leakage. Enabling ray logging (enable_ray_logging=True). Specify `ds_args={'enable_ray_logging': False}` if you experience logging issues.
2025-02-25 10:00:53,066	INFO worker.py:1810 -- Started a local Ray instance. View the dashboard at http://127.0.0.1:8265/ 
		Context path: "/content/agModels-predictWear1.1/ds_sub_fit/sub_fit_ho"
(_dystack pid=2871) Running DyStack sub-fit ...
(_dystack pid=2871) Beginning AutoGluon training ... Time limit = 867s
(_dystack pid=2871) AutoGluon will save models to "/content/agModels-predictWear1.1/ds_sub_fit/sub_fit_ho"
(_dystack pid=2871) Train Data Rows:    155259
(_dystack pid=2871) Train Data Columns: 7
(_dystack pid=2871) Label Column:       Wear(Micron)
(_dystack pid=2871) Problem Type:       regression
(_dystack pid=2871) Preprocessing data ...
(_dystack pid=2871) Using Feature Generators to preprocess the data ...
(_dystack pid=2871) Fitting AutoMLPipelineFeatureGenerator...
(_dystack pid=2871) 	Available Memory:                    11262.41 MB
(_dystack pid=2871) 	Train Data (Original)  Memory Usage: 8.29 MB (0.1% of available memory)
(_dystack pid=2871) 	Inferring data type of each feature based on column values. Set feature_metadata_in to manually specify special dtypes of the features.
(_dystack pid=2871) 	Stage 1 Generators:
(_dystack pid=2871) 		Fitting AsTypeFeatureGenerator...
(_dystack pid=2871) 			Note: Converting 1 features to boolean dtype as they only contain 2 unique values.
(_dystack pid=2871) 	Stage 2 Generators:
(_dystack pid=2871) 		Fitting FillNaFeatureGenerator...
(_dystack pid=2871) 	Stage 3 Generators:
(_dystack pid=2871) 		Fitting IdentityFeatureGenerator...
(_dystack pid=2871) 	Stage 4 Generators:
(_dystack pid=2871) 		Fitting DropUniqueFeatureGenerator...
(_dystack pid=2871) 	Stage 5 Generators:
(_dystack pid=2871) 		Fitting DropDuplicatesFeatureGenerator...
(_dystack pid=2871) 	Types of features in original data (raw dtype, special dtypes):
(_dystack pid=2871) 		('float', []) : 2 | ['Sliding Distace(m)', 'load (N)']
(_dystack pid=2871) 		('int', [])   : 5 | ['Time(secs)', 'alloy', 'grid size', 'speed(rpm)', 'sliding distance (s)']
(_dystack pid=2871) 	Types of features in processed data (raw dtype, special dtypes):
(_dystack pid=2871) 		('float', [])     : 2 | ['Sliding Distace(m)', 'load (N)']
(_dystack pid=2871) 		('int', [])       : 4 | ['Time(secs)', 'grid size', 'speed(rpm)', 'sliding distance (s)']
(_dystack pid=2871) 		('int', ['bool']) : 1 | ['alloy']
(_dystack pid=2871) 	0.3s = Fit runtime
(_dystack pid=2871) 	7 features in original data used to generate 7 features in processed data.
(_dystack pid=2871) 	Train Data (Processed) Memory Usage: 7.26 MB (0.1% of available memory)
(_dystack pid=2871) Data preprocessing and feature engineering runtime = 0.35s ...
(_dystack pid=2871) AutoGluon will gauge predictive performance using evaluation metric: 'root_mean_squared_error'
(_dystack pid=2871) 	This metric's sign has been flipped to adhere to being higher_is_better. The metric score can be multiplied by -1 to get the metric value.
(_dystack pid=2871) 	To change this, specify the eval_metric parameter of Predictor()
(_dystack pid=2871) Large model count detected (112 configs) ... Only displaying the first 3 models of each family. To see all, set `verbosity=3`.
(_dystack pid=2871) User-specified model hyperparameters to be fit:
(_dystack pid=2871) {
(_dystack pid=2871) 	'NN_TORCH': [{}, {'activation': 'elu', 'dropout_prob': 0.10077639529843717, 'hidden_size': 108, 'learning_rate': 0.002735937344002146, 'num_layers': 4, 'use_batchnorm': True, 'weight_decay': 1.356433327634438e-12, 'ag_args': {'name_suffix': '_r79', 'priority': -2}}, {'activation': 'elu', 'dropout_prob': 0.11897478034205347, 'hidden_size': 213, 'learning_rate': 0.0010474382260641949, 'num_layers': 4, 'use_batchnorm': False, 'weight_decay': 5.594471067786272e-10, 'ag_args': {'name_suffix': '_r22', 'priority': -7}}],
(_dystack pid=2871) 	'GBM': [{'extra_trees': True, 'ag_args': {'name_suffix': 'XT'}}, {}, {'learning_rate': 0.03, 'num_leaves': 128, 'feature_fraction': 0.9, 'min_data_in_leaf': 3, 'ag_args': {'name_suffix': 'Large', 'priority': 0, 'hyperparameter_tune_kwargs': None}}],
(_dystack pid=2871) 	'CAT': [{}, {'depth': 6, 'grow_policy': 'SymmetricTree', 'l2_leaf_reg': 2.1542798306067823, 'learning_rate': 0.06864209415792857, 'max_ctr_complexity': 4, 'one_hot_max_size': 10, 'ag_args': {'name_suffix': '_r177', 'priority': -1}}, {'depth': 8, 'grow_policy': 'Depthwise', 'l2_leaf_reg': 2.7997999596449104, 'learning_rate': 0.031375015734637225, 'max_ctr_complexity': 2, 'one_hot_max_size': 3, 'ag_args': {'name_suffix': '_r9', 'priority': -5}}],
(_dystack pid=2871) 	'XGB': [{}, {'colsample_bytree': 0.6917311125174739, 'enable_categorical': False, 'learning_rate': 0.018063876087523967, 'max_depth': 10, 'min_child_weight': 0.6028633586934382, 'ag_args': {'name_suffix': '_r33', 'priority': -8}}, {'colsample_bytree': 0.6628423832084077, 'enable_categorical': False, 'learning_rate': 0.08775715546881824, 'max_depth': 5, 'min_child_weight': 0.6294123374222513, 'ag_args': {'name_suffix': '_r89', 'priority': -16}}],
(_dystack pid=2871) 	'FASTAI': [{}, {'bs': 256, 'emb_drop': 0.5411770367537934, 'epochs': 43, 'layers': [800, 400], 'lr': 0.01519848858318159, 'ps': 0.23782946566604385, 'ag_args': {'name_suffix': '_r191', 'priority': -4}}, {'bs': 2048, 'emb_drop': 0.05070411322605811, 'epochs': 29, 'layers': [200, 100], 'lr': 0.08974235041576624, 'ps': 0.10393466140748028, 'ag_args': {'name_suffix': '_r102', 'priority': -11}}],
(_dystack pid=2871) 	'RF': [{'criterion': 'gini', 'ag_args': {'name_suffix': 'Gini', 'problem_types': ['binary', 'multiclass']}}, {'criterion': 'entropy', 'ag_args': {'name_suffix': 'Entr', 'problem_types': ['binary', 'multiclass']}}, {'criterion': 'squared_error', 'ag_args': {'name_suffix': 'MSE', 'problem_types': ['regression', 'quantile']}}],
(_dystack pid=2871) 	'XT': [{'criterion': 'gini', 'ag_args': {'name_suffix': 'Gini', 'problem_types': ['binary', 'multiclass']}}, {'criterion': 'entropy', 'ag_args': {'name_suffix': 'Entr', 'problem_types': ['binary', 'multiclass']}}, {'criterion': 'squared_error', 'ag_args': {'name_suffix': 'MSE', 'problem_types': ['regression', 'quantile']}}],
(_dystack pid=2871) 	'KNN': [{'weights': 'uniform', 'ag_args': {'name_suffix': 'Unif'}}, {'weights': 'distance', 'ag_args': {'name_suffix': 'Dist'}}],
(_dystack pid=2871) }
(_dystack pid=2871) AutoGluon will fit 2 stack levels (L1 to L2) ...
(_dystack pid=2871) Fitting 108 L1 models, fit_strategy="sequential" ...
(_dystack pid=2871) Fitting model: KNeighborsUnif_BAG_L1 ... Training model for up to 577.72s of the 866.79s of remaining time.
(_dystack pid=2871) 	-419.9658	 = Validation score   (-root_mean_squared_error)
(_dystack pid=2871) 	0.76s	 = Training   runtime
(_dystack pid=2871) 	5.46s	 = Validation runtime
(_dystack pid=2871) Fitting model: KNeighborsDist_BAG_L1 ... Training model for up to 565.33s of the 854.39s of remaining time.
(_dystack pid=2871) 	-439.7896	 = Validation score   (-root_mean_squared_error)
(_dystack pid=2871) 	0.48s	 = Training   runtime
(_dystack pid=2871) 	5.01s	 = Validation runtime
(_dystack pid=2871) Fitting model: LightGBMXT_BAG_L1 ... Training model for up to 559.75s of the 848.82s of remaining time.
(_dystack pid=2871) 	Fitting 8 child models (S1F1 - S1F8) | Fitting with ParallelLocalFoldFittingStrategy (2 workers, per: cpus=1, gpus=0, memory=0.51%)
(_ray_fit pid=3216) /usr/local/lib/python3.11/dist-packages/dask/dataframe/__init__.py:42: FutureWarning: 
(_ray_fit pid=3216) Dask dataframe query planning is disabled because dask-expr is not installed.
(_ray_fit pid=3216) 
(_ray_fit pid=3216) You can install it with `pip install dask[dataframe]` or `conda install dask`.
(_ray_fit pid=3216) This will raise in a future version.

the code when fitting any model with ParallelLocalFoldFittingStrategy  is using dask dataframes, so if it got it in the dependencies, it would much better in this subcase

**Comment by Innixma** [comment]
I'm guessing that you installed dask via some other dependency in your environment. Do you still see the warning if you create a fresh python environment and only install AutoGluon in it?

As it stands in this PR, I don't think the change would stop this warning from occurring. We don't use dask in AutoGluon. Seems more likely a problem with dask and ray. I believe the warning is harmless and shouldn't be a blocker.

**Comment by Innixma** [comment]
Closing this PR as the proposed change wouldn't address the mentioned issue, and dask is not a dependency of AutoGluon.
